sql/log.cc

```diff
@@ -5687,20 +5687,39 @@ THD::binlog_start_trans_and_stmt()
     Ha_trx_info *ha_info;
     ha_info= this->ha_data[binlog_hton->slot].ha_info + (mstmt_mode ? 1 : 0);
 
-    if (!ha_info->is_started() && wsrep_gtid_mode
-            && this->variables.gtid_seq_no)
+    if (!ha_info->is_started())
     {
       binlog_cache_mngr *const cache_mngr=
         (binlog_cache_mngr*) thd_get_ha_data(this, binlog_hton);
       binlog_cache_data *cache_data= cache_mngr->get_binlog_cache_data(1);
       IO_CACHE *file= &cache_data->cache_log;
       Log_event_writer writer(file, cache_data);
+
+      if (wsrep_gtid_mode && this->variables.gtid_seq_no)
+      {
         Gtid_log_event gtid_event(this, this->variables.gtid_seq_no,
                             this->variables.gtid_domain_id,
                             true, LOG_EVENT_SUPPRESS_USE_F,
                             true, 0);
-      gtid_event.server_id= this->variables.server_id;
-      writer.write(&gtid_event);
+        gtid_event.server_id= this->variables.server_id;
+        writer.write(&gtid_event);
+      }
+
+      if (this->transaction.xid_state.xa_state == XA_ACTIVE)
+      {
+        Format_description_log_event fd_ev(4);
+        fd_ev.checksum_alg= BINLOG_CHECKSUM_ALG_OFF; //(enum_binlog_checksum_alg)binlog_checksum_options;
```

Comment:
Remove commented code. Why disable checksum alg here ? If this is not going to be written on binlog (even when log-bin is enabled) add a comment.